propertyestimator/backends/dask.py

```diff
@@ -1,13 +1,228 @@
 """
 A collection of property estimator compute backends which use dask as the distribution engine.
 """
+import importlib
 import logging
 import multiprocessing
 import os
+import shutil
 
 from dask import distributed
+from distributed import get_worker
+from simtk import unit
 
-from .backends import PropertyEstimatorBackend, ComputeResources
+from propertyestimator.workflow.plugins import available_protocols
+from .backends import PropertyEstimatorBackend, ComputeResources, QueueWorkerResources
+
+
+class DaskLSFBackend(PropertyEstimatorBackend):
+    """A property estimator backend which uses a dask-jobqueue `LSFCluster`
+    objects to run calculations within an existing LSF queue.
+    """
+
+    def __init__(self,
+                 minimum_number_of_workers=1,
+                 maximum_number_of_workers=1,
+                 resources_per_worker=QueueWorkerResources(),
+                 default_memory_unit=unit.giga*unit.byte,
+                 queue_name='default',
+                 extra_script_commands=None,
+                 adaptive_interval='10000ms'):
+
+        """Constructs a new DaskLocalClusterBackend
+
+        Parameters
+        ----------
+        minimum_number_of_workers: int
+            The minimum number of workers to request from the queue system.
+        maximum_number_of_workers: int
+            The maximum number of workers to request from the queue system.
+        resources_per_worker: QueueWorkerResources
+            The resources to request per worker.
+        default_memory_unit: simtk.Unit
+            The default unit used by the LSF queuing system when
+            defining memory usage limits / requirements - this
+            must be compatible with `unit.bytes`.
+        queue_name: str
+            The name of the queue which the workers will be requested
+            from.
+        extra_script_commands: list of str
```

Comment:
I'd suggest the name `setup_script_commands`. It's probably more intuitive when you're familiar with tests and/or `timeit`.